src/java/org/apache/cassandra/tools/JMXTool.java

```diff
@@ -0,0 +1,838 @@
+                                                              child.setType(INFO_TYPE.getType());
+                                                              return super.constructObject(child);
+                                                          }));
+                }
+                else
+                {
+                    return super.constructObject(node);
+                }
+            }
+        }
+    }
+
+    private static Map<String, Info> load(JMXServiceURL url) throws IOException, MalformedObjectNameException, IntrospectionException, InstanceNotFoundException, ReflectionException
+    {
+        try (JMXConnector jmxc = JMXConnectorFactory.connect(url, null))
+        {
+            MBeanServerConnection mbsc = jmxc.getMBeanServerConnection();
+
+            Map<String, Info> map = new TreeMap<>();
+            for (String pkg : new TreeSet<>(METRIC_PACKAGES))
+            {
+                Set<ObjectName> metricNames = new TreeSet<>(mbsc.queryNames(new ObjectName(pkg + ":*"), null));
+                for (ObjectName name : metricNames)
+                {
+                    if (mbsc.isRegistered(name))
+                    {
+                        MBeanInfo info = mbsc.getMBeanInfo(name);
+                        map.put(name.toString(), Info.from(info));
+                    }
+                }
+            }
+            return map;
+        }
+    }
+
+    private static String getAccess(MBeanAttributeInfo a)
+    {
+        String access;
+        if (a.isReadable())
+        {
+            if (a.isWritable())
+                access = "read/write";
+            else
+                access = "read-only";
+        }
+        else if (a.isWritable())
+            access = "write-only";
+        else
+            access = "no-access";
+        return access;
+    }
+
+    private static String normalizeType(String type)
+    {
+        switch (type)
+        {
+            case "[Z":
+                return "boolean[]";
+            case "[B":
+                return "byte[]";
+            case "[S":
+                return "short[]";
+            case "[I":
+                return "int[]";
+            case "[J":
+                return "long[]";
+            case "[F":
+                return "float[]";
+            case "[D":
+                return "double[]";
+            case "[C":
+                return "char[]";
+        }
+        if (type.startsWith("[L"))
+            return type.substring(2, type.length() - 1) + "[]"; // -1 will remove the ; at the end
+        return type;
+    }
+
+    private static final StringBuilder ROW_BUFFER = new StringBuilder();
+
+    private static void printRow(PrintStream out, String... args)
+    {
+        ROW_BUFFER.setLength(0);
+        ROW_BUFFER.append("\t\t");
+        for (String a : args)
+            ROW_BUFFER.append(a).append("\t");
+        out.println(ROW_BUFFER);
+    }
+
+    public static final class Info
```

Comment:
Are notifications considered part of this? The `StreamManager` exposes some.

Reply:
at the moment no, thought it simplest to add later; would be to just add that to info